packages/phone-number-privacy/src/database/wrappers/number-pairs.ts

```diff
@@ -8,13 +9,18 @@ export async function getNumberPairContacts(
   userPhone: string,
   contactPhones: string[]
 ): Promise<string[]> {
-  const contentPairs = await getDatabase()(NUMBER_PAIRS_TABLE)
-    .select(NUMBER_PAIRS_COLUMN.userPhoneHash)
-    .where(NUMBER_PAIRS_COLUMN.contactPhoneHash, userPhone)
+  try {
+    const contentPairs = await getDatabase()(NUMBER_PAIRS_TABLE)
+      .select(NUMBER_PAIRS_COLUMN.userPhoneHash)
+      .where(NUMBER_PAIRS_COLUMN.contactPhoneHash, userPhone)
 
-  return contentPairs
-    .map((contractPair) => contractPair[NUMBER_PAIRS_COLUMN.userPhoneHash])
-    .filter((number) => contactPhones.includes(number))
+    return contentPairs
+      .map((contractPair) => contractPair[NUMBER_PAIRS_COLUMN.userPhoneHash])
+      .filter((number) => contactPhones.includes(number))
```

Comment:
`.filter((number) => contactPhones.includes(number))`
This is O(n^2). Each number (N) is compared to contactPhones (N) via a linear search. If we build a map/object and use that instead, the lookup will be constant.

Reply:
oh that's a good point -- i should change it to the js equivalent of a hashset

Reply:
@aslawson in JS one would typically use either an object `{}` or a map `new Map<type1, type2>()`, whichever you prefer.